nmproxy/nm-proxy.go

```diff
@@ -13,13 +14,14 @@ import (
 )
 
 // Start - setups the global cfg for proxy and starts the proxy server
-func Start(ctx context.Context, mgmChan chan *nm_models.HostPeerUpdate, stunAddr string, stunPort int) {
+func Start(ctx context.Context, wg *sync.WaitGroup, mgmChan chan *nm_models.HostPeerUpdate, stunAddr string, stunPort int) {
 
 	if config.GetCfg().IsProxyRunning() {
```

Comment:
if you move the wg.Add to below this check, the check for if IsProxyRunning in functions/daemon.go:84 becomes redundant does it not?

Reply:
that check is required in functions/daemon.go otherwise context can get overwritten

Reply:
okay